static/src/javascripts/projects/commercial/modules/comscore.spec.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import config from 'lib/config';
+import { init, _ } from './comscore';
+
+const getComscoreScripTags = (): NodeList<HTMLElement> =>
+    document.querySelectorAll('script#comscore');
+
+const isValidComscoreScriptTag = (tag: HTMLElement) =>
+    tag instanceof HTMLScriptElement &&
+    tag.src === _.comscoreSrc &&
+    tag.async === true;
+
+describe('comscore init', () => {
+    beforeEach(() => {
+        config.set('switches.comscore', true);
+    });
+
+    it('should do nothing if the comscore switch is off', () => {
+        config.set('switches.comscore', false);
+        init();
+
+        const comscoreTags = getComscoreScripTags();
+        expect(comscoreTags.length).toBe(0);
+    });
+
+    it('should drop the corect script tag if the comscore switch is on', () => {
+        config.set('switches.comscore', true);
+        init();
+
+        const comscoreTags = getComscoreScripTags();
+        expect(comscoreTags.length).toBe(1);
+        expect(isValidComscoreScriptTag(comscoreTags[0])).toBe(true);
+    });
+});
+
+describe('comscore getGlobals', () => {
+    it('returns an object with the correct comscorekw variable set', () => {
+        const randomStr = Math.random()
```

Comment:
Consider using fast-check when you want to write tests with random inputs. It's much more powerful and flexible then writing random stuff in directly, it will run the test multiple times to check it against lots of data before it passes, and it will even do clever tricks to try and find the minimal case that fails (to improve the failure message).

I'm not 100% sure fast-check is included in frontend yet so no dramas here, but I mention it as an FYI.

An example Guardian codebase that uses fast check is consent logs, which I think you've worked on a little:
e.g. https://github.com/guardian/consent-logs/blob/master/src/validation.spec.ts

Reply:
fast-check is not included in frontend and I don't feel it's necessary to add it just for this case. But looking at this again I do realise there's no point in testing a single random string so I'll change it to test a fixed string instead.